[php] Specify when config env variable should be available (#3952)
DataDog/documentation: content/tracing/advanced_usage/_index.md

Debug mode is disabled by default. To enable it, set the environment variable `DD_TRACE_DEBUG=true`.

> *An important note*: If you use code auto-instrumentation (the recommended approach) please be aware that the

Reviewer: ```suggestion
*An important note*: If you use code auto-instrumentation (the recommended approach) be aware that the
```

Author: Thanks :)

> instrumenting code is executed before any user code. So the environment variables below must be set at the server

Reviewer: ```suggestion
instrumenting code is executed before any user code. As a result, the environment variables below must be set at the server
```

> level and be available to the PHP runtime before any user code is executed. So, for example, `putenv()` and `.env`

Reviewer: ```suggestion
level and be available to the PHP runtime before any user code is executed. For example, `putenv()` and `.env`
```

files would not work.

```php

> DD_TRACE_DEBUG=true

Reviewer 2: For the examples, it might be good to include the main suggested ways to add the environment variables depending on your server. Since we only support Apache, FPM & CLI server, we could have a short blurb and example of each. :)

### Apache

Set using [`SetEnv`](https://httpd.apache.org/docs/2.4/mod/mod_env.html#setenv) from the server config, virtual host, directory, or **.htaccess** file.

```
SetEnv DD_TRACE_DEBUG true
```

### nginx

Set using [`fastcgi_param`](http://nginx.org/en/docs/http/ngx_http_fastcgi_module.html#fastcgi_param) from the `http`, `server`, or `location` contexts.

```
fastcgi_param DD_TRACE_DEBUG true;
```

### PHP CLI server

Set in the command line to start the server.

```
DD_TRACE_DEBUG=true php -S localhost:8888
```

Author: Good idea @SammyK , will add it :)
